test-data/unit/check-custom-plugin.test

```diff
@@ -0,0 +1,71 @@
+-- Test cases for user-defined plugins
+--
+-- Note: Plugins used by tests live under test-data/unit/plugins. Defining
+--       plugin files in test cases does not work reliably.
```

Comment:
Not sure exactly. I spent some time trying to get it to work and then gave up. A couple of hypotheses:
* If two test cases use the same plugin name, we need to reliably free the plugin between test cases as otherwise the plugin module will be visible between tests. It may be hard to ensure that the plugin is always deleted at the end of the test case.
* Python may do some caching / time stamp magic that may cause the import machinery to sometimes not see a recently created module.